### PR TITLE
feat(prover): add nitro local mode, test script, and stack overflow fix

### PR DESCRIPTION
**Pull request by 0x00101010**
## Summary

- Restructure prover CLI into `nitro server`, `nitro enclave`, and `nitro local` subcommands, replacing the old proxy-based TEE server
- Add `nitro local` mode that runs server + enclave in a single process for development without AWS Nitro hardware
- Add `test-nitro-prover.sh` script that fetches live L1/L2 data via `optimism_outputAtBlock` and sends `prover_prove` requests to a locally running prover
- Fix stack overflow in the derivation pipeline by `Box::pin`-ing the client future (deeply nested async state machine exceeded the 2MB default stack)
- Enable `--enable-experimental-witness-endpoint` in the `nitro-local` Justfile recipe
- Cap L1 head at current chain tip to prevent null hash errors when proving recent blocks
- Remove unused proxy and transport config modules from `crates/proof/tee/server/`

## Testing

Tested locally against Base mainnet using `just tee nitro-local` and `test-nitro-prover.sh`. The prover successfully advances through the derivation pipeline (stack overflow fixed, witness endpoint enabled). Verified that L1 head capping prevents failures when proving blocks near the chain tip.

### CODE REVIEW

**Comment by cb-heimdall** [comment]
<h1>✅ Heimdall Review Status</h1>
<table>
<tr>
<th>Requirement</th>
<th>Status</th>
<th>More Info</th>
</tr>
<tr>
<td>Reviews</td>
<td>
✅
<code>1/1</code>
</td>
<td>
<details>
<summary>Denominator calculation</summary>
<table>
<tr>
<td>
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><code>1</code> if user is bot</td>
<td>0</td>
</tr>
<tr>
<td><code>1</code> if user is external</td>
<td>0</td>
</tr>
<tr>
<td><code>2</code> if repo is sensitive</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>.codeflow.yml</code>
</td>
<td>1</td>
</tr>
<tr>
<td>
Additional review requirements
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><strong>Max</strong></td>
<td><strong>0</strong></td>
</tr>
</table>
</details>
</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>CODEOWNERS</code>
</td>
<td>0</td>
</tr>
<tr>
<td>Global minimum</td>
<td>0</td>
</tr>
<tr>
<td><strong>Max</strong></td>
<td>
<strong>
1
</strong>
</td>
</tr>
</table>
</details>
</td>
<td>1</td>
</tr>
<tr>
<td><code>1</code> if commit is unverified</td>
<td>0</td>
</tr>
<tr>
<td><strong>Sum</strong></td>
<td><strong>1</strong></td>
</tr>
</table>
</details>
</td>
</tr>
</table>

File: bin/prover/src/cli.rs [hunk 1]
```diff
@@ -10,38 +26,254 @@ pub(crate) struct Cli {
...
+        let transport = Arc::new(NativeTransport::new({
+            let server = Arc::clone(&enclave_server);
+            move |preimages| {
+                let server = Arc::clone(&server);
+                let preimages = preimages.to_vec();
+                tokio::task::block_in_place(|| {
+                    tokio::runtime::Handle::current()
+                        .block_on(server.prove(preimages))
+                        .expect("enclave prove failed")
+                })
```
[the remark this answers is not in the document]
Comment (0x00101010) [reply bin/prover/src/cli.rs:245]: This is test code

File: bin/prover/README.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # `base-prover`
 
-TEE prover server binary with two subcommands:
+TEE prover binary for AWS Nitro Enclaves with two subcommands:
```
[the remark this answers is not in the document]
Comment (0x00101010) [reply bin/prover/README.md]: done

File: crates/proof/tee/server/src/proxy.rs [hunk 3]
```diff
@@ -1,197 +0,0 @@
...
-        }
-    }
-
-    if response.is_empty() || !clean {
-        None
-    } else {
-        pool.put(conn);
-        Some(response)
-    }
-}
```
[the remark this answers is not in the document]
Comment (0x00101010) [reply crates/proof/tee/server/src/proxy.rs:197]: done

File: bin/prover/src/cli.rs [hunk 4]
```diff
@@ -1,6 +1,22 @@
...
+use alloy_primitives::{Address, B256};
+use base_consensus_registry::Registry;
+use base_proof_host::ProverConfig;
+#[cfg(target_os = "linux")]
+use base_proof_tee_nitro::{NitroEnclave, EnclaveConfig};
+use base_proof_tee_nitro::NitroProverServer;
+#[cfg(feature = "local")]
+use base_proof_tee_nitro::Server;
+#[cfg(feature = "local")]
+use base_proof_tee_nitro::EnclaveConfig;
```
[the remark this answers is not in the document]
Comment (0x00101010) [reply bin/prover/src/cli.rs]: done

File: Cargo.lock [hunk 5]
```diff
@@ -3186,6 +3186,7 @@ dependencies = [
 ]
 
 [[package]]
+<<<<<<< Updated upstream
```
[the remark this answers is not in the document]
Comment (0x00101010) [reply Cargo.lock]: done

File: bin/prover/src/cli.rs [hunk 6]
```diff
@@ -1,6 +1,22 @@
...
+use base_proof_transport::NativeTransport;
```
[the remark this answers is not in the document]
Comment (0x00101010) [reply bin/prover/src/cli.rs]: done

File: bin/prover/src/cli.rs [hunk 7]
```diff
@@ -10,38 +27,254 @@ pub(crate) struct Cli {
...
+        let enclave_server = Arc::new(Server::new(&enclave_config)?);
+        let transport = Arc::new(NativeTransport::new({
+            let server = Arc::clone(&enclave_server);
+            move |preimages| {
+                let server = Arc::clone(&server);
+                let preimages = preimages.to_vec();
+                tokio::task::block_in_place(|| {
+                    tokio::runtime::Handle::current()
+                        .block_on(server.prove(preimages))
+                        .expect("enclave prove failed")
```
Comment (leopoldjoy) [inline bin/prover/src/cli.rs]: Nit: Even for local dev, replacing `.expect(...)` with proper error propagation would give a more readable error on failure instead of a panic + backtrace. Minor though since this is test code.

File: etc/scripts/local/test-nitro-prover.sh [hunk 8]
```diff
@@ -0,0 +1,164 @@
...
+
+# ---------------------------------------------------------------------------
+# Step 1: Determine the L2 block range to prove
+#   claimed = safe head (already fully derived, so all L1 batch data exists)
+#   agreed  = safe head - 10 (the prover re-derives 10 blocks)
+# ---------------------------------------------------------------------------
+sync_status=$(rpc "$OP_NODE_URL" "optimism_syncStatus" "[]")
+safe_l2_number=$(echo "$sync_status" | jq -r '.safe_l2.number')
+CLAIMED_L2_BLOCK_NUMBER=$((safe_l2_number))
+AGREED_L2_BLOCK_NUMBER=$((CLAIMED_L2_BLOCK_NUMBER - 10))
```
Comment (leopoldjoy) [inline etc/scripts/local/test-nitro-prover.sh]: Nit: The 10-block range is hardcoded. Consider making it a `BLOCK_RANGE` env var for flexibility (e.g., `BLOCK_RANGE="${BLOCK_RANGE:-10}"`).

Comment (0x00101010) [reply etc/scripts/local/test-nitro-prover.sh]: done

File: bin/prover/src/cli.rs (hunk 7 above)
Comment (0x00101010) [reply bin/prover/src/cli.rs]: done

File: bin/prover/src/cli.rs [hunk 9]
```diff
@@ -10,38 +27,254 @@ pub(crate) struct Cli {
...
+        let enclave_server = Arc::new(Server::new(&enclave_config)?);
+        let transport = Arc::new(NativeTransport::new({
+            let server = Arc::clone(&enclave_server);
+            move |preimages| {
+                let server = Arc::clone(&server);
+                let preimages = preimages.to_vec();
+                tokio::task::block_in_place(|| {
+                    tokio::runtime::Handle::current()
+                        .block_on(server.prove(preimages))
+                        .map_err(|e| TransportError::Codec(e.to_string()))
```
[the remark this answers is not in the document]
Comment (0x00101010) [reply bin/prover/src/cli.rs]: done

File: bin/prover/src/cli.rs [hunk 10]
```diff
@@ -10,38 +27,254 @@ pub(crate) struct Cli {
...
+    config_hash: B256,
+
+    /// Expected PCR0 measurement of the enclave image.
+    #[arg(long, env = "TEE_IMAGE_HASH")]
+    tee_image_hash: B256,
+
+    /// Enable experimental `debug_executePayload` witness endpoint.
+    #[arg(long, env = "ENABLE_EXPERIMENTAL_WITNESS_ENDPOINT")]
+    enable_experimental_witness_endpoint: bool,
+}
```
[the remark this answers is not in the document]
Comment (0x00101010) [reply bin/prover/src/cli.rs:215]: done

File: bin/prover/src/cli.rs [hunk 11]
```diff
@@ -10,38 +27,240 @@ pub(crate) struct Cli {
...
+#[cfg(feature = "local")]
+impl NitroLocalArgs {
+    async fn run(self) -> eyre::Result<()> {
+        let rollup_config = Registry::rollup_config(self.server.l2_chain_id)
+            .ok_or_else(|| eyre!("unknown L2 chain ID: {}", self.server.l2_chain_id))?
+            .clone();
+
+        let l1_config = Registry::l1_config(rollup_config.l1_chain_id)
+            .ok_or_else(|| eyre!("unknown L1 chain ID: {}", rollup_config.l1_chain_id))?
+            .clone();
```
[the remark this answers is not in the document]
Comment (0x00101010) [reply bin/prover/src/cli.rs:226]: do as a follow up

**Review by leopoldjoy — APPROVED** [review]
